### ISSUE TITLE
React component crashes react when loaded in a display:none div.

### ISSUE THREAD

**Issue by LyraelRayne** [Microsoft/BotFramework-WebChat#515]
I have a React application which shows/hides the webchat when a user clicks its header. By default the application starts hidden (i.e. the parent of the Chat has display:none).

A few seconds after the page finishes loading I get this

```
TypeError: Cannot read property 'offsetWidth' of null
HistoryView.componentDidUpdate
D:\webstuff\application\node_modules\botframework-webchat\built\History.js:36
  33 | // Measure the message padding by subtracting the known large width
  34 | var paddedWidth = measurePaddedWidth(this.carouselActivity.messageDiv) - this.largeWidth;
  35 | // Subtract the padding from the offsetParent's width to get the width of the content
> 36 | var maxContentWidth = this.carouselActivity.messageDiv.offsetParent.offsetWidth - paddedWidth;
  37 | // Subtract the content width from the chat width to get the margin.
  38 | // Next time we need to get the content width (on a resize) we can use this margin to get the maximum content width
  39 | var carouselMargin = this.props.size.width - maxContentWidth;
```

This does not happen if the application starts up with the chat showing, even if display:none is set afterwards.

Currently I work around it by doing 
```
componentDidMount() {
         this.onLoadTimer = setTimeout(() => this.setState({isShowing: false}), 1);
    }
```

but obviously this is not preferred.

**Comment by danmarshall** [comment]
The component needs to do some measuring when it is initialized. Perhaps you can refrain from initializing it until you are ready to display it?

**Comment by LyraelRayne** [comment]
I can probably make the render of the JSX component conditional on the user having selected to show it at least once, but it'd be a little awkward and I'd probably only consider that a workaround not a fix. It'd be a great improvement to the robustness of the component if it didn't crash the entire application when the measurements failed (perhaps logging to the console that it detected a failure state).

I presume that those measurements need to be repeated each time the parent is resized anyway. It'd be nice if the component quietly accepted that it can't do its measurements when hidden, awaiting the time when the parent is resized (in this case "shown") and it can repeat the process.

**Comment by nicekiwi** [comment]
+1

**Comment by RaoVenka** [comment]
We created bot using  .net code and hosted on cloud platform. Please find below the link
http://sonataharmonysite.azurewebsites.net/#

Web Chat is not working in internet explorer 11 browser.Could you please help me on this.
Please find below error screenshot.

![image](https://user-images.githubusercontent.com/28774238/29156589-5c578ce8-7dc0-11e7-8f43-7b3a22788acd.png)

**Comment by danmarshall** [comment]
Hello all, the guidance is to use CSS `visibility:hidden` instead of `display:none` until further notice. We would like to fix this issue, but we will not be able to get to it soon, so please try the workaround and let us know. Thanks :)

**Comment by compulim** [comment]
Fixed in #1003.
